app/views/course/assessment/answer/programming/_programming.json.jbuilder

```diff
@@ -99,7 +101,9 @@ json.explanation do
       json.failureType 'private_test'
     end
 
-    json.correct attempt&.auto_grading && attempt&.correct
+    passed_evaluation_tests = failed_test_cases_by_type['evaluation_test'].blank?
+
+    json.correct attempt&.auto_grading && attempt&.correct && (can_grade && passed_evaluation_tests)
```

Comment:
Actually, I realise if a student accesses the page, then `json.correct` becomes `false`? @purfectliterature

Reply:
Right, I meant to add `can_grade ? passed_evaluation_tests : true` to not change anything for students, and only consider evaluation tests for graders.

I will make another PR for this.